jicoco-kotlin/src/main/kotlin/org/jitsi/config/LegacyConfigurationServiceShim.kt

```diff
@@ -0,0 +1,217 @@
+        return getOrDefault(defaultValue) {
+            legacyShimConfig.getterFor(String::class)(path)
+        }
+    }
+
+    override fun getProperty(path: String?): Any? {
+        path ?: return null
+        return getOrDefault(null) {
+            legacyShimConfig.getterFor(ConfigObject::class)(path)
+        }
+    }
+
+    override fun addPropertyChangeListener(listener: PropertyChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun addPropertyChangeListener(propertyName: String?, listener: PropertyChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun addVetoableChangeListener(listener: ConfigVetoableChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun addVetoableChangeListener(propertyName: String?, listener: ConfigVetoableChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun removePropertyChangeListener(listener: PropertyChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun removePropertyChangeListener(propertyName: String?, listener: PropertyChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun removeVetoableChangeListener(listener: ConfigVetoableChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun removeVetoableChangeListener(propertyName: String?, listener: ConfigVetoableChangeListener?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun getAllPropertyNames(): MutableList<String> {
+        throw Exception("Unsupported")
+    }
+
+    override fun getPropertyNamesByPrefix(prefix: String?, exactPrefixMatch: Boolean): MutableList<String> {
+        prefix ?: return mutableListOf()
+        return getOrDefault(mutableListOf()) {
+            val obj = legacyShimConfig.getterFor(ConfigObject::class)(prefix)
+            obj.toConfig().entrySet()
+                    .map { it.key }
+                    .filter { propName ->
+                        if (exactPrefixMatch) {
+                            // If an exact prefix match was requested, don't
+                            // include keys nested under further scopes
+                            !propName.contains(".")
+                        }
+                        true
+                    }
+                    .map { "$prefix.$it"}
+                    .toMutableList()
+        }
+    }
+
+    override fun getPropertyNamesBySuffix(suffix: String?): MutableList<String> {
+        throw Exception("Unsupported")
+    }
+
+    override fun logConfigurationProperties(excludePatter: String?) {
+        // Leaving this as a no-op, we already log the config
+    }
+
+    override fun reloadConfiguration() {
+        legacyShimConfig.reload()
+    }
+
+    override fun removeProperty(propertyName: String?) {
+        throw Exception("Unsupported")
+    }
+
+    override fun setProperties(properties: MutableMap<String, Any>?) {
```

Comment:
We might need this one (and some others) for jigasi. I guess we'll find out when we try to update jicoco there.

Reply:
Actually, jigasi would be using the old config through jitsi/libjitsi, right?

Reply:
yup, exactly.